File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let answer = [];
    let temp = {};
    
    for(let i=0; i<s.length; i++){
        if(temp[s[i]] === undefined){
            temp[s[i]] = i;
            answer.push(-1);
        }else{
            answer.push(i - temp[s[i]]);
            temp[s[i]] = i;
        }
    }
    return answer
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        
        int[] answer = new int[s.length()];
        HashMap<Character, Integer> map = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char cur = s.charAt(i);
            if (!map.containsKey(cur)) {
                map.put(cur, i);
                answer[i] = -1;
            } else {
                answer[i] = i - map.get(cur);
                map.put(cur, i);
            }
        }
        
        return answer;
    }
}